runtime/Cpp/runtime/src/atn/PredictionContext.h

```diff
@@ -10,243 +10,263 @@
-     *
-     * <p><embed src="images/FullMerge_SameRoot.svg" type="image/svg+xml"/></p>
-     *
-     * @param a the first {@link SingletonPredictionContext}
-     * @param b the second {@link SingletonPredictionContext}
-     * @param rootIsWildcard {@code true} if this is a local-context merge,
-     * otherwise false to indicate a full-context merge
-     */
-    static Ref<PredictionContext> mergeRoot(const Ref<SingletonPredictionContext> &a,
-      const Ref<SingletonPredictionContext> &b, bool rootIsWildcard);
-
-    /**
-     * Merge two {@link ArrayPredictionContext} instances.
-     *
-     * <p>Different tops, different parents.<br>
-     * <embed src="images/ArrayMerge_DiffTopDiffPar.svg" type="image/svg+xml"/></p>
-     *
-     * <p>Shared top, same parents.<br>
-     * <embed src="images/ArrayMerge_ShareTopSamePar.svg" type="image/svg+xml"/></p>
-     *
-     * <p>Shared top, different parents.<br>
-     * <embed src="images/ArrayMerge_ShareTopDiffPar.svg" type="image/svg+xml"/></p>
-     *
-     * <p>Shared top, all shared parents.<br>
-     * <embed src="images/ArrayMerge_ShareTopSharePar.svg" type="image/svg+xml"/></p>
-     *
-     * <p>Equal tops, merge parents and reduce top to
-     * {@link SingletonPredictionContext}.<br>
-     * <embed src="images/ArrayMerge_EqualTop.svg" type="image/svg+xml"/></p>
-     */
-    static Ref<PredictionContext> mergeArrays(const Ref<ArrayPredictionContext> &a,
-      const Ref<ArrayPredictionContext> &b, bool rootIsWildcard, PredictionContextMergeCache *mergeCache);
-
-  protected:
-    /// Make pass over all M parents; merge any equal() ones.
-    /// @returns true if the list has been changed (i.e. duplicates where found).
-    static bool combineCommonParents(std::vector<Ref<PredictionContext>> &parents);
-
-  public:
-    static std::string toDOTString(const Ref<PredictionContext> &context);
-
-    static Ref<PredictionContext> getCachedContext(const Ref<PredictionContext> &context,
-      PredictionContextCache &contextCache,
-      std::map<Ref<PredictionContext>, Ref<PredictionContext>> &visited);
-
-    // ter's recursive version of Sam's getAllNodes()
-    static std::vector<Ref<PredictionContext>> getAllContextNodes(const Ref<PredictionContext> &context);
-    static void getAllContextNodes_(const Ref<PredictionContext> &context,
-      std::vector<Ref<PredictionContext>> &nodes, std::set<PredictionContext *> &visited);
-
-    virtual std::string toString() const;
-    virtual std::string toString(Recognizer *recog) const;
-
-    std::vector<std::string> toStrings(Recognizer *recognizer, int currentState);
-    std::vector<std::string> toStrings(Recognizer *recognizer, const Ref<PredictionContext> &stop, int currentState);
-  };
-
-  struct PredictionContextHasher {
-    size_t operator () (const Ref<PredictionContext> &k) const {
-      return k->hashCode();
+    namespace atn {
+
+        struct PredictionContextHasher;
```

Comment:
Looks like you indented the atn namespace block. This was not so before by intention. Please revert that.